plugins/ptp_operator/config/config.go

```diff
@@ -0,0 +1,216 @@
+	if _, found := l.EventThreshold[iface]; found {
+		l.lock.Lock()
+		defer l.lock.Unlock()
+		delete(l.EventThreshold, iface)
+	}
+}
+
+// UpdatePTPThreshold .. update ptp threshold
+func (l *LinuxPTPConfUpdate) UpdatePTPThreshold() {
+	for _, profile := range l.NodeProfiles {
+		if _, found := l.EventThreshold[*profile.Interface]; !found {
+			l.SetDefaultPTPThreshold(*profile.Interface)
+		}
+		if t, found := l.EventThreshold[*profile.Interface]; found {
+			if profile.PtpClockThreshold.MaxOffsetThreshold != 0 {
+				t.MaxOffsetThreshold = profile.PtpClockThreshold.MaxOffsetThreshold
+			}
+			if profile.PtpClockThreshold.MinOffsetThreshold != 0 {
+				t.MinOffsetThreshold = profile.PtpClockThreshold.MinOffsetThreshold
+			}
+			if profile.PtpClockThreshold.HoldOverTimeout != 0 {
+				t.HoldOverTimeout = profile.PtpClockThreshold.HoldOverTimeout
+			}
+			log.Infof("update ptp threshold values for %s\n holdoverTimeout: %d\n maxThreshold: %d\n minThreshold: %d\n",
+				*profile.Interface,t.HoldOverTimeout, t.MaxOffsetThreshold, t.MinOffsetThreshold)
+		}
+	}
+}
+
+// UpdateConfig ... update profile
+func (l *LinuxPTPConfUpdate) UpdateConfig(nodeProfilesJSON []byte) error {
+	if string(l.appliedNodeProfileJSON) == string(nodeProfilesJSON) {
+		return nil
+	}
+
+	if nodeProfiles, ok := tryToLoadConfig(nodeProfilesJSON); ok {
+		log.Info("load ptp profiles")
+		l.appliedNodeProfileJSON = nodeProfilesJSON
+		l.NodeProfiles = nodeProfiles
+		l.UpdateCh <- true
+		return nil
+	}
+
+	if nodeProfiles, ok := tryToLoadOldConfig(nodeProfilesJSON); ok {
+		// Support empty old config
+		// '{"name":null,"interface":null}'
+		if nodeProfiles[0].Name == nil || nodeProfiles[0].Interface == nil {
+			log.Infof("Skip no profile %+v", nodeProfiles[0])
+			return nil
+		}
+
+		log.Info("load profiles using old method")
+		l.appliedNodeProfileJSON = nodeProfilesJSON
+		l.NodeProfiles = nodeProfiles
+		l.UpdateCh <- true
+
+		return nil
+	}
+
+	return fmt.Errorf("unable to load profile config")
+}
+
+// Try to load the multiple policy config
+func tryToLoadConfig(nodeProfilesJSON []byte) ([]PtpProfile, bool) {
+	var ptpConfig []PtpProfile
+	err := json.Unmarshal(nodeProfilesJSON, &ptpConfig)
+	if err != nil {
+		return nil, false
+	}
+
+	return ptpConfig, true
+}
+
+// For backward compatibility we also try to load the one policy scenario
+func tryToLoadOldConfig(nodeProfilesJSON []byte) ([]PtpProfile, bool) {
+	ptpConfig := &PtpProfile{}
+	err := json.Unmarshal(nodeProfilesJSON, ptpConfig)
+	if err != nil {
+		return nil, false
+	}
+
+	return []PtpProfile{*ptpConfig}, true
+}
+
+// WatchConfigUpdate watch for ptp config update
+func (l *LinuxPTPConfUpdate) WatchConfigUpdate(nodeName string, closeCh chan struct{}) {
+	tickerPull := time.NewTicker(time.Duration(l.intervalUpdate) * time.Second)
+	defer tickerPull.Stop()
+	for {
+		select {
+		case <-tickerPull.C:
+			log.Infof("ticker pull for ptp profile updates")
```

Comment:
Should be `log.Info` with no arguments. (or should we use `log.debug`?)

Reply:
looks like our linter is broken, Usually, the linter should catch this type. I will fix this

Reply:
Fixed